### PR TITLE
Capture async stack traces of scheduler jobs

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This adds annotations that help [IntelliJ capture async stack traces](https://www.jetbrains.com/help/idea/debug-asynchronous-code.html).
Debugging locally becomes much easier as it is now visible where an actor job is coming from.

Before:
image

After: 
image

### CODE REVIEW

**Comment by npepinpe** [comment]
Cool, wasn't aware of this! :+1: 

On one hand, I wish we had this in prod, on the other hand I imagine it would have quite a performance hit :smile:

**Review by npepinpe — APPROVED** [review]

**Comment by oleschoenburg** [comment]
> On one hand, I wish we had this in prod, on the other hand I imagine it would have quite a performance hit 😄

Yeah, I wish :sweat_smile: 

I'll try and backport this too.

bors merge

File: scheduler/src/main/java/io/camunda/zeebe/scheduler/ActorJob.java [hunk 1]
```diff
@@ -38,6 +39,7 @@ public void onJobAddedToTask(final ActorTask task) {
     schedulingState = TaskSchedulingState.QUEUED;
   }
 
+  @Async.Execute
```
Comment (npepinpe) [inline scheduler/src/main/java/io/camunda/zeebe/scheduler/ActorJob.java:42]: :question: How is the job mapped back with the key? i.e. `@Async.Schedule` uses the job as key. How does this know which stack to use? :D

Comment (oleschoenburg) [reply scheduler/src/main/java/io/camunda/zeebe/scheduler/ActorJob.java:42]: It's implicitly "this" which work nicely here :tada: See https://www.jetbrains.com/help/idea/debug-asynchronous-code.html#bc2ffa26

Comment (npepinpe) [reply scheduler/src/main/java/io/camunda/zeebe/scheduler/ActorJob.java:42]: Oh, right, `@Async.Execute` is on the `ActorJob` itself. Gotcha. Cool :+1:

**Pull request merged**